Snowflake.Data.Tests/ChunkParserFactoryTest.cs

```diff
@@ -0,0 +1,83 @@
+﻿/*
+ * Copyright (c) 2021 Snowflake Computing Inc. All rights reserved.
+ */
+
+namespace Snowflake.Data.Tests
+{
+    using NUnit.Framework;
+    using Snowflake.Data.Configuration;
+    using Snowflake.Data.Core;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Security;
+    using System.Text;
+    using System.Threading;
+
+    [TestFixture]
+    class ChunkParserFactoryTest
+    {
+        bool UseV2JsonParserDefault = SFConfiguration.Instance().UseV2JsonParser;
+        int ChunkParserVersionDefault = SFConfiguration.Instance().GetChunkParserVersion();
+
+        [TearDown]
+        public void AfterTest()
+        {
+            SFConfiguration.Instance().UseV2JsonParser = UseV2JsonParserDefault; // Return to default version
+            SFConfiguration.Instance().ChunkParserVersion = ChunkParserVersionDefault; // Return to default version
+        }
+
+        [Test]
+        [Ignore("ChunkParserFactoryTest")]
+        public void ChunkParserFactoryTestDone()
+        {
+            // Do nothing;
+        }
+
+        [Test]
+        public void TestGetParser([Values(false, true)] bool useV2JsonParser, [Values(1, 2, 3, 4)] int chunkParserVersion)
+        {
+            // Set configuration settings
+            SFConfiguration.Instance().UseV2JsonParser = useV2JsonParser;
+            SFConfiguration.Instance().ChunkParserVersion = chunkParserVersion;
+
+            // Get parser using sample stream
+            byte[] bytes = Encoding.UTF8.GetBytes("test");
+            Stream stream = new MemoryStream(bytes);
+
+            IChunkParser parser = null;
+
+            // GetParser() throws an error when ChunkParserVersion is not 1-3
```

Comment:
Unless we can finally remove from the code all those parser versions, my further idea would be to introduce an enum to be reused within the code to get rid of magic numbers of parser/downloader versions.

Reply:
Yeah that sounds good, making an enum will help make sure the version is within the proper range